### PR TITLE
(1/3) RUM-392 create core testFixtures source set

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

This PR adds a `testFixtures` source set to the Core module, and adds test fixture classes that will be shared with other modues (in the next PR)

### Motivation

This is part of our effort on improving our test suites across the mobile teams

### CODE REVIEW

File: dd-sdk-android-core/build.gradle.kts [hunk 1]
```diff
@@ -170,9 +170,11 @@ dependencies {
     testImplementation(libs.bundles.testTools)
     unmock(libs.robolectric)
 
-    // Static Analysis
-    // TODO MTG-12 detekt(project(":tools:detekt"))
-    // TODO MTG-12 detekt(libs.detektCli)
+    // Test Fixtures
+    testFixturesImplementation(libs.kotlin)
```
Comment (0xnm) [inline dd-sdk-android-core/build.gradle.kts:174]: why it is needed? `implementation` dependencies don't cover `testFixtures` source set (like they cover `test` source set)?

File: dd-sdk-android-core/build.gradle.kts [hunk 2]
```diff
@@ -170,9 +170,11 @@ dependencies {
...
+    testFixturesImplementation(libs.bundles.jUnit5)
+    testFixturesImplementation(libs.okHttp)
```
Comment (0xnm) [inline dd-sdk-android-core/build.gradle.kts:176]: it seems no class in `testFixtures` source set is using it, why do we need it?

**Review by 0xnm — APPROVED** [review]
just two minor questions, otherwise lgtm.

**Review by mariusc83 — COMMENTED** [review]
Nice ;)

**Review by mariusc83 — APPROVED** [review]

File: dd-sdk-android-core/build.gradle.kts (hunk 1 above)
Comment (xgouchet) [reply dd-sdk-android-core/build.gradle.kts:174]: For Gradle, no it's not necessary but AndroidStudio needed that to be happy.

File: dd-sdk-android-core/build.gradle.kts (hunk 2 above)
Comment (xgouchet) [reply dd-sdk-android-core/build.gradle.kts:176]: it's needed for something that's going to be added next but not in this PR yet

**Pull request merged**